src/Agent.Worker/Handlers/Handler.cs

```diff
@@ -298,24 +300,65 @@ protected void AddPrependPathToEnvironment()
             }
         }
 
-        protected void RemovePSModulePathFromEnvironment()
+        [SupportedOSPlatform("windows")]
+        protected bool PsModulePathContainsPowershellCoreLocations()
         {
-            if (AgentKnobs.CleanupPSModules.GetValue(ExecutionContext).AsBoolean() && PlatformUtil.RunningOnWindows)
+            bool checkLocationsKnob = AgentKnobs.CheckPsModulesLocations.GetValue(HostContext).AsBoolean();
+
+            bool isPwshCore = Inputs.TryGetValue("pwsh", out string pwsh) && StringUtil.ConvertToBoolean(pwsh);
+
+            if (!PlatformUtil.RunningOnWindows || !checkLocationsKnob || isPwshCore)
             {
-                bool useInteropKnob = AgentKnobs.UseInteropToFindParentProcess.GetValue(ExecutionContext).AsBoolean();
+                return false;
+            }
+
+            const string PSModulePath = nameof(PSModulePath);
+
+            bool localVariableExists = Environment.TryGetValue(PSModulePath, out string localVariable);
+            bool localVariableContainsPwshLocations = PsModulePathUtil.ContainsPowershellCoreLocations(localVariable);
```

Comment:
[not blocking, minor] using value of localVariable even though TryGetValue returned false.  Probably not an issue

Reply:
There's a check for null in the util method, so it is probably fine